spec.js

```diff
@@ -0,0 +1,66 @@
+const assert = require('chai').assert;
+const {
+  expect
+} = require('chai');
+
+describe('Scholastic App', function () {
+  
+    beforeEach(function() {
+      browser.ignoreSychronization = true;
+      return browser.manage().window().maximize();
+    })
+  
+  it('should add item to cart', async function () {
+    try {    
+      await browser.get('https://shop.scholastic.com/teachers-ecommerce/teacher/tsohomepage.html');
+      await browser.waitForAngularEnabled(false);
+      //hover on programm. books and libraries
+      await browser.actions().mouseMove(element(by.xpath('//a[@data-element-linkname="h-nav_programs-books-and-libraries"]'))).perform();
+      await browser.wait(ExpectedConditions.visibilityOf(element(by.css('.horizontal-dd'))), 6000);
+      //click on "English Learners" from Classrom Libraries
+      await element(by.xpath('//div[@class="horizontal-dd"]//a[@data-element-linkname="h-nav_programs-books-and-libraries_classroom-libraries_english-learners"]')).click();
+      await browser.sleep(3000);
+      //click on Filters Grade
+      await element(by.xpath("//div[@id='cio-search-facets']//div[contains(text(),'Grade')]")).click();     
+      await browser.sleep(1000);
+      //click on Grade 12 checkbox
+      const gradeTwelve = await element(by.xpath("//div[@class='leftNav section']//input[@id='grade-grade-12']"));
+      await browser.executeScript('arguments[0].click()', gradeTwelve);
+      await browser.sleep(1000); 
+      const resultsElement = await element(by.css('.cio-num-results'));
+      await browser.wait(ExpectedConditions.textToBePresentInElement(resultsElement, '19 Results', 5000));
```

Comment:
It's a bad idea to hardcode values here (`19 Results`)

It's better to save text before filtering (or whatever you did) and wait for text to be changed after filtering